### DISCUSSION TITLE
Rework of the "Decimal Point" Logic  in 7Seg UI

### DISCUSSION THREAD

**Discussion by pizman** [MobiFlight/MobiFlight-Connector#642] [Ideas]
Hi
I just want to start a Brain Storming about the DP in the 7Seg System   ( That is currently "hardcoded" )

Lets say we like to blank a Display like the AP-ALT .....  We make a Compare and Say "If xxx=True then Spaces, else $"
If a Display have a Decimal Point ( Preselected by the UI) like a COM1 Radio 123,456   then this not work....
With the "Spaces" within the Main Config the Display show  "_ _ _, _ _ _"  So the DP remains.

Only workarround a.t.m. is the "Old Logic" to Build TWO Configs.... One with Value and DP  the Other Without DP and always spaces..... and then we need preconditions in each Config.....  For sure it work but its not comfortable.
Also related to the know BUG  that we need always use the 100% same Diggits if 2 Configs are related to same Diggits.
(E.g. Blank 8 Diggits witch one Config....  while the Diggits are normaly used by 2 other Configs is not possible)

*************
Idea:
Why we not simply say that a "Space" set the WHOLE Diggit to Blank ?
Currently we only blank the Segments, But the DP Bit is not involved and MF looks like send it still in case of the UI Checkmark.

So.... If i send a Space for the diggit where a DP is selected THIS should overwrite the former DP temporarly!

*************

Additional Idea from Stephan and myself.

Maybe we could add something new in the "Bright Setting" Tool that comes into MF with 9.1
Here we can choose by a value of 0-15 for a Brightness Setting of 1/32 to 31/23 ......   BUT the lowest Value is just Dimmed but not OFF.
What do you think about if we would say "IF Birhgt Value =0  THEN  Not send the Register to "0000" and dimm 1/32.....  Send in that case simply " BLANK  ( including DP) for all  8 Diggits of That Display! 

Result:  We would no longer need to make Spaces Compares.....   If a Display must be blank we would simply set one Global Config to "0" and all related 7Seg Configs with the "Dimm Tool enabled" get Blank including DP.

Usecase :  E.g. If Battery Main Switch is OFF then Indicator = 0   ELSE Indicater is "15" .

**Comment by DocMoebiuz** [comment]
i think it would be best to have a way to use decimal numbers natively. I think the only thing that would be needed is the alignment of the decimal point at a certain digit... but i am not sure about that.

**Reply by DocMoebiuz** [reply]
this would require to remove the limitation that we only can use integer numbers

**Reply by pizman** [reply]
> I think the only thing that would be needed is the alignment of the decimal point at a certain digit... but i am not sure about that.

In theory Yes....  But do we have a Situation that occure ?
E.g. the Coms....    Right. The Decimal must be always at the middle Diggit like 123,456 
In Theory if the Value would be just 98,123 the decimal is moving....
BUT   All possible frequencys are Above 100 ....  So the Decimal can not Change.

Where we get a Problem is e.g. in the FPA (Flight path angle)  A Value between 1,0, and 5,0  Degrees with one Decimal.
Here 3,5 work....   But 3,0 would be shown as "3" pretty sure. 
May a "tricky workaround like $+0.01 is needed.... So we always show e.g. 3.001 (while the 01 is not displayed in case of missing Diggits)

**Reply by pizman** [reply]
UPDATE:
After a nice Brainstorming with @GioCC  we get an idea....
![grafik](https://user-images.githubusercontent.com/26087116/151838923-461d9530-cb25-4f7c-8321-321cf5107363.png)

Instead of "Set all Segments to "0" ( what can collide with our Hardcoded DP and maybe a resending of the still active Configs)
we could use the "Shutdown" Feature.
If i understand right, the Max remains working..... Means in theory also new Values are send to it..... It just no longer show something on the Displays ( including the DecimalPoints)

So i think about a Feature....  IF Dimm Setting = 0  THEN send Shoutdown Command.... ELSE if DIM >0 then send the specific Dim Value to the Register as we du at the moment !
